drivers/gpio/gpio_stellaris.c

```diff
@@ -0,0 +1,434 @@
+/*
+ * Copyright (c) 2018 Zilogic Systems.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <errno.h>
+#include <device.h>
+#include <gpio.h>
+#include <soc.h>
+#include <sys_io.h>
+#include "gpio_utils.h"
+#include "gpio_stellaris.h"
+
+#define DEV_CFG(dev)                                     \
+	((const struct gpio_stellaris_config *const)     \
+	(dev)->config->config_info)
+
+#define DEV_DATA(dev)					 \
+	((struct gpio_stellaris_runtime *const)          \
+	(dev)->driver_data)
+
+#define GPIO_REG_ADDR(base, offset) (base + offset)
+
+#define GPIO_RW_ADDR(base, offset, p)			 \
+	(GPIO_REG_ADDR(base, offset) | (1 << (p + 2)))
+
+enum gpio_ports {
+	GPIO_PORT_A,
+	GPIO_PORT_B,
+	GPIO_PORT_C,
+	GPIO_PORT_D,
+	GPIO_PORT_E,
+	GPIO_PORT_F,
+	GPIO_PORT_G,
+};
+
+enum gpio_regs {
+	GPIO_DATA_OFFSET = 0x000,
+	GPIO_DIR_OFFSET = 0x400,
+	GPIO_DEN_OFFSET = 0x51C,
+	GPIO_IS_OFFSET = 0x404,
+	GPIO_IBE_OFFSET = 0x408,
+	GPIO_IEV_OFFSET = 0x40C,
+	GPIO_IM_OFFSET = 0x410,
+	GPIO_MIS_OFFSET = 0x418,
+	GPIO_ICR_OFFSET = 0x41C,
+};
+
+static u32_t portx_pins[] = {0xFF, 0xFF, 0xFF,
```

Comment:
It looks like the only reason for port in gpio_stellaris_config is to lookup portx_pins[], so why not just put portx_pins in gpio_stellaris_config, and remove gpio_ports

Reply:
moved inside config structure